source/src/enet_encap/encap.c

```diff
@@ -578,7 +578,7 @@ EipStatus HandleReceivedInvalidCommand(const EncapsulationData *const receive_da
  *                      kInvalidSession .. no free session available
  */
 int GetFreeSessionIndex(void) {
-  for(size_t session_index = 0; session_index < OPENER_NUMBER_OF_SUPPORTED_SESSIONS; session_index++) {
+  for(int session_index = 0; session_index < OPENER_NUMBER_OF_SUPPORTED_SESSIONS; session_index++) {
```

Comment:
what's the issue here? the comparison with OPENER_NUMBER_OF_SUPPORTED_SESSIONS?
In that case making the number an unsigned, by adding U to the preprocessor constant would be the better solution imho